### DISCUSSION TITLE
Full-fledged Pawn Status for Custom Pawns

### DISCUSSION THREAD

**Discussion by rapiermother** [fairy-stockfish/Fairy-Stockfish#672] [Ideas]
Using `pawnTypes`, we can add new pieces which behave like pawns in almost evey way; as of now, the `pawnTypes` PieceSet allows pawn-style promotion, en passant capturing, and n-move rule considerations for such pieces.

In order to grant custom pieces "full-fledged" pawn powers however, they'll need to be able to double-step/triple-step. My hope is that either `pawnTypes` can be updated such that all members of its PieceSet can do so, or that a `doubleStepTypes` option can be added to facilitate this.

**Comment by ianfab** [comment]
Did you check the "initial moves" in the Betza notation like this:
https://github.com/fairy-stockfish/Fairy-Stockfish/blob/8549ed1dd01e641e234e073a0b86dac70923dcf4/src/variants.ini#L299-L304

With this you can't get two independent regions like for double- and triple-steps but only of them, but having both should be a very special case anyway, so I think that is acceptable to be restricted to standard pawns.

**Reply by rapiermother** [reply]
I can confirm that adding the 'ifmnD' to the Betza of my custom pawn types did the charm as you've suggested. Thanks very much!

**Comment by rapiermother** [comment]
See @ianfab's comment on the OP for the direct solution.
